### DISCUSSION TITLE
Create codemeta as accessible from the API

### DISCUSSION THREAD

**Discussion by drmowinckels** [r-universe-org/help#237] [Ideas]
Codemeta is slowly growing up, and might (hopefully) help create more FAIR software.

https://codemeta.github.io/

There are two R packages that will auto-create codemeta.json based on repo/R package files.

https://github.com/ropensci/codemetar
https://github.com/cboettig/codemeta

While we cannot force developers to adopt these things, we might be able to help them by making it part of the R-universe eco-system.

My, very naïve, idea is that the build process could:
- use codemeta.json if it exists, OR
- create codemeta.json with one of the above packs if not existing

And then expose this json as an API endpoint.

This would make the software metadata easily accessible for funders or other stakeholders or those who would like to make software-crawlers to help people find software that might aid them in their work in a more agnostic way.

**Comment by jeroen** [comment]
Thanks for the suggestion.

R-universe of course all about the metadata (and we have a lot of it), so we could expose that any formats. Most metadata suggested by codemeta are already in our database, so we don't need to scrape it again using the codemeta(r) packages. 

One thing that is not clear to me is how we need to expose this for consumers to find this data. We actually have been maintaining a [codemetar registry for ropensci packages](https://github.com/ropensci/roregistry/blob/gh-pages/registry.json) for a long time, but I am not sure anyone is using that data. I also can't find information on this on the codemeta website. Are there good examples of popular codemeta data providers?

**Comment by drmowinckels** [comment]
Excellent questions! I'm asking around people I've met regarding this and I'll get back to you once I have some answers :)

**Reply by drmowinckels** [reply]
So far, what I know is that Software Heritage is at least looking for it from GitHub and the like. So does HAL (https://hal.science/)  though I'm not quite sure how or from where.
